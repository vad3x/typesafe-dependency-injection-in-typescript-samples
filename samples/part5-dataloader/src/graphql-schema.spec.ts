import { GraphQLSchema, graphql } from "graphql";
import { FindOrderById } from "./find-order-by-id";
import { graphqlSchema } from "./graphql-schema";

describe("#graphqlSchema", () => {
  const findOrderByIdMock = jest.fn<
    ReturnType<FindOrderById>,
    Parameters<FindOrderById>
  >();

  afterEach(() => {
    jest.resetAllMocks();
  });

  test("should create registry registration with graphqlSchema field", () => {
    // arrange
    const fn = graphqlSchema();

    // act
    const reg = fn({ findOrderById: findOrderByIdMock });

    // assert
    expect(reg).toStrictEqual({
      graphqlSchema: expect.any(GraphQLSchema),
    });
  });

  describe("#graphqlSchema", () => {
    const fn = graphqlSchema();
    const reg = fn({ findOrderById: findOrderByIdMock });

    test("should query Query.order(id: _)", async () => {
      // arrange
      findOrderByIdMock.mockResolvedValue({
        id: "order-id0",
      });

      // act
      const result = await graphql({
        schema: reg.graphqlSchema,
        source: /* GraphQL */ `
          query {
            order(id: "order-id0") {
              id
            }
          }
        `,
      });

      // assert
      expect(result).toMatchSnapshot("graphql-result");

      expect(findOrderByIdMock).toBeCalledTimes(1);
      expect(findOrderByIdMock).toBeCalledWith("order-id0");
    });
  });
});
