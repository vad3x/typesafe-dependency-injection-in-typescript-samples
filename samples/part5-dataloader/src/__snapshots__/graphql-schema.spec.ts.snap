// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#graphqlSchema #graphqlSchema should query Query.order(id: _): graphql-result 1`] = `
{
  "data": {
    "order": {
      "id": "order-id0",
    },
  },
}
`;
